Add Dactyl Manuform board (variants) (#761)
KMKfw/kmk_firmware: boards/dactyl_manuform/readme.md

# Dactyl Manuform

The [Dactyl Manuform](https://github.com/tshort/dactyl-keyboard) is a handwired, split bodied, tented, curved keywell keyboard - adapted from the [Dactyl](https://github.com/adereth/dactyl-keyboard) keyboard with the thumb cluster design of the [ManuForm](https://geekhack.org/index.php?topic=46015.0) keyboard. Information for building a Dactyl Manuform found in first link.

![Imgur](https://i.imgur.com/7y0Vbyd.jpg)
*Two wired Dactyl Manuform 4x6 variants*

## Variants

Dactyl Manuform's are built in variations that cater for different row and column counts, and thumb clusters.  

> V ariants are denoted as `RowCount`*x*`ColumnCount` and share the common configuration of:

Reviewer: ```suggestion
Variants are denoted as `RowCount`*x*`ColumnCount` and share the common configuration of:
```

- The finger key-well bottom row has 2 keys, 1 each in ring and middle columns. Exception to this rule is the 7 column variants that have two additional keys in this row.

> - The thumb cluster has 6 keys, arranged in 2 columns by 3 rows .

Reviewer: ```suggestion
- The thumb cluster has 6 keys, arranged in 2 columns by 3 rows.
```

# Dactyl Manuform

> The [Dactyl Manuform](https://github.com/tshort/dactyl-keyboard) is a handwired, split bodied, tented, curved keywell keyboard - adapted from the [Dactyl](https://github.com/adereth/dactyl-keyboard) keyboard with the thumb cluster design of the [ManuForm](https://geekhack.org/index.php?topic=46015.0) keyboard. Information for building a Dactyl Manuform found in first link.

Reviewer: Break up long lines to preferable 80 characters.
External links should have the current date attached (i.e.: last known valid date), if there's the potential that the link may change withing a couple of years. (Yes, that caused issues in the past3)

Author: Hi @xs5871,

looking at other readme.md files I'm not seeing a date attached to external links. Could you give example of how date should be added?

Reviewer: Yeah, sadly we never finished the doc linting and doc guidlines.
"[text (YYYY-MM)](link)" should do. Not strictly necessary for expected long-lived external links; github should be fine without, unless it's a small personal project. I wasn't familiar with geekhack, but they seem to be very long lived as well, although that could benefit from a date of post creation.
The imgur could state roughly when the image was taken, for example.

Author: date added for ManuForm link

![Imgur](https://i.imgur.com/7y0Vbyd.jpg)
*Two wired Dactyl Manuform 4x6 variants*

## Variants

Dactyl Manuform's are built in variations that cater for different row and column counts, and thumb clusters.  

> 

Reviewer: ```suggestion
```

> As standard: 
> - The finger keywell bottom row has 2 keys, 1 each in ring and middle columns
>     - Exception to this rule is the 7 column variants, that have two additional keys in this row
> - The thumb cluster has 6 keys, arranged in 2 columns by 3 rows 

Reviewer: ```suggestion
V ariants are denoted as `RowCount`*x*`ColumnCount` and share the common configuration of:
- The finger key-well bottom row has 2 keys, 1 each in ring and middle columns. Exception to this rule is the 7 column variants that have two additional keys in this row.
- The thumb cluster has 6 keys, arranged in 2 columns by 3 rows .
```

Is "keywell" the proper spelling? It looks like a compositum that should have a hyphen, but I'm not a linguist.

> 
> Variants are denoted as `RowCount`*x*`ColumnCount`

Reviewer: ```suggestion
```

### Row
| Count | Description |
| --- | --- |

> | 4 | Three rows, typically for alphabet and some puncuation characters, with 2 key (finger keywell) bottom row |
> | 5 | As *4 row* with number row above |
> | 6 | As *5 row* with function row above |

Reviewer: ```suggestion
| 4 | Three rows, typically for alphabet and some punctuation characters, with 2 key (finger keywell) bottom row |
| 5 | Like *4 rows* with number row above |
| 6 | Like *5 rows* with function row above |
```

### Column
| Count | Description |
| --- | --- |

> | 5 | A column for each finger with additional column for first finger |  
> | 6 | As *5 column* with additional pinky finger column |
> | 7 | As *6 column* with either an additional first finger column (`5x7`) or additional pinky column (`6x7`) |

Reviewer: ```suggestion
| 5 | A column for each finger with additiona index finger column |  
| 6 | Like *5 columns* with additional pinky finger column |
| 7 | Like *6 columns* with either an additional index finger column (`5x7`) or additional pinky column (`6x7`) |
```

> ## Microcontroller support

Reviewer: ```suggestion
## Microcontroller Support
```

> Amend `controller` in the following line of desired variant's `kb.py` file to supported microcontroller listed in `kmk/quickpin/pro_micro`:

Reviewer: If you have to edit it, what's the point of having 6 different `kb.py`s in the first place? Why not a single one for 4x5 with the additional rows/colums commented out?
Edit: I guess the `coord_mapping` would be an issue, althogh most of that could be alleviated by reordering the column pins. New request: reorder the column pins to make the `coord_map` as linear as possible.

Author: Sorry @xs5871 , I have no idea what entails.

My understanding is the `coord_mapping` is directly related to pin matrix, thus there is a different coord_mapping per keyboard as their pin matrices are unique.

Would you happen to have a reference to your request?

Reviewer: No, sorry, strike that. Didn't consider that these are all split boards. Let's leave it the way it is.

```python
from kmk.quickpin.pro_micro.controller import pinout as pins
```
For example, nice!nano controller(s): 
```python
from kmk.quickpin.pro_micro.nice_nano import pinout as pins
```

## KMK Specifics

Extentions enabled by default:  
- [Layers](/docs/en/layers.md): As many as you want/need

> - [Split](/docs/en/split_keyboards.md): Configured to 1 wire UART to match legacy configuration. Please see documentation for enabling 2 wire UART or, for capable controllers, Bluetooth

Reviewer: ```suggestion
- [Split](/docs/en/split_keyboards.md): Configured to 1-wire UART to match legacy configuration. Please see documentation for enabling 2-wire UART or, for capable controllers, Bluetooth.
```

Again, split long lines.
